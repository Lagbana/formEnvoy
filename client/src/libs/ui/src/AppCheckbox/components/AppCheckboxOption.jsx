import React from 'react';
import  {AppIcon}  from '../../AppIcon';
import { TouchableOpacity, View, StyleSheet, Text } from 'react-native';
import { Colors, Typography } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppCheckboxOptionType } from '../AppCheckbox.interface';
import { AppText } from '../../AppText';

export interface AppCheckboxOptionProps<T> {
	option: AppCheckboxOptionType<T>;
	checked?: boolean;
	onPress: (option: AppCheckboxOptionType<T>) => void;
	scheme?: 'light' | 'dark';
	isHtml?: boolean;
	layout?: {
    spacing?: number;
    checkboxSize?: number;
    fontSize?: number;
  };
}

export function AppCheckboxOption<T = any>({ option, checked, onPress, layout, scheme, isHtml }: AppCheckboxOptionProps<T>) {

	const styles = StyleSheet.create({
    item: {
      marginVertical: layout?.spacing ?? 12,
      flexDirection: 'row' as 'row',
      alignItems: 'center' as 'center',
    },
    checkbox: {
      width: layout?.checkboxSize ?? 26,
      height: layout?.checkboxSize ?? 26,
      borderRadius: 6,
      marginRight: layout?.checkboxSize ?? 25,
      backgroundColor: scheme === 'light' ? Colors.white : 'transparent',
      borderColor: Colors.purpleLight,
      borderWidth: 2,
    },
    selectedCheckbox: {
      backgroundColor: Colors.green,
      borderWidth: 0,
    },
    icon: {
      margin: 2,
      color: Colors.white,
    },
    label: {
      ...Typography.labelDefault,
      color: scheme === 'light' ? Colors.purple : Colors.white + '88',
      marginTop: 1,
      flex: 1,
    },
    htmlLabel: {
      fontSize: layout?.fontSize ?? 16,
			lineHeight: (layout?.fontSize ?? 16) * 1.5,
      flex: 1,
      marginBottom: 0,
    },
    box: {},
	});
	
	const handlePress = () => onPress(option);

	const label = (isHtml) ? <AppText style={styles.htmlLabel}>{option.label}</AppText> : 
		<Text style={[styles.label, checked ? {} : {}]}>{option.label}</Text>;

	return (
    <TouchableOpacity
      style={styles.item}
      onPress={handlePress}
      onLongPress={handlePress}
    >
      <View style={[styles.checkbox, checked ? styles.selectedCheckbox : {}]}>
        {checked && (
          <AppIcon
            name="check"
            size={layout?.checkboxSize ? layout.checkboxSize * 0.85 : 22}
            style={styles.icon}
          />
        )}
      </View>
      {label}
		</TouchableOpacity>
	);

}