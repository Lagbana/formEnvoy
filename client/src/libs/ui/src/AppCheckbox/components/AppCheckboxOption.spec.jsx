import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppCheckboxOption } from './AppCheckboxOption'

configure({ adapter: new Adapter() })

test('<AppCheckboxOption /> renders correctly', () => {
	const component = shallow(<AppCheckboxOption
		option={{ label: '', value: '' }}
		onPress={(option = { label: '', value: ''}) => {}}
	/>)
  expect(toJson(component)).toMatchSnapshot()
})
