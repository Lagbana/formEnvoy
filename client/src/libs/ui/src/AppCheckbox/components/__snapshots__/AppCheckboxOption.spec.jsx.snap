// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppCheckboxOption /> renders correctly 1`] = `
<ForwardRef
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 12,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#CACADA",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 26,
          "marginRight": 25,
          "width": 26,
        },
        Object {},
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF88",
          "flex": 1,
          "fontFamily": "ObjektivMk1-Bold",
          "fontSize": 18,
          "lineHeight": 19.8,
          "marginTop": 1,
          "paddingTop": 4.5,
        },
        Object {},
      ]
    }
  />
</ForwardRef>
`;
